src/sage/matrix/matrix_polynomial_dense.pyx

```diff
@@ -4079,3 +4084,564 @@ cdef class Matrix_polynomial_dense(Matrix_generic_dense):
+            sage: rcomp = mat._basis_completion_via_reversed_approx(); rcomp
+            [        5*x^2 + 4*x + 1             5*x^2 + 2*x                   5*x^2]
+            [          2*x^3 + 4*x^2 2*x^3 + 6*x^2 + 2*x + 1       2*x^3 + x^2 + 3*x]
+            sage: basis = mat.stack(rcomp); basis
+            [      x^3 + 4*x^2 + 2*x   x^3 + 5*x^2 + 5*x + 4   x^3 + 6*x^2 + 4*x + 6]
+            [        5*x^2 + 4*x + 1             5*x^2 + 2*x                   5*x^2]
+            [          2*x^3 + 4*x^2 2*x^3 + 6*x^2 + 2*x + 1       2*x^3 + x^2 + 3*x]
+            sage: basis.determinant()
+            6
+
+        The following matrix has rank `2` and trivial Smith form. It can be
+        completed row-wise into a `3 \times 3` unimodular matrix::
+
+            sage: mat = matrix(ring, 2, 3, \
+                    [[x^2 + 5*x + 5,   3*x^2 + x + 3, 4*x^2 + 5*x + 4], \
+                     [5*x^2 + 4*x,   3*x^2 + 4*x + 5, 5*x^2 + 5*x + 3]])
+            sage: rcomp = mat._basis_completion_via_reversed_approx(); rcomp
+            [  2*x^2 + 1 4*x^2 + 3*x 2*x^2 + 3*x]
+            sage: mat.stack(rcomp).determinant()
+            3
+
+        The following matrix has rank 1 and its nonzero Smith factor is `x+3`.
+        A row-wise completion has a single nonzero row, whereas a column-wise
+        completion has two columns; in both cases, the Smith form is preserved::
+
+            sage: mat = matrix(ring, 3, 2, \
+                    [[    x^3 + x^2 + 5*x + 5,         2*x^3 + 2*x + 4], \
+                     [  3*x^3 + 2*x^2 + x + 3,   6*x^3 + 5*x^2 + x + 1], \
+                     [2*x^3 + 5*x^2 + 3*x + 4, 4*x^3 + 6*x^2 + 5*x + 6]])
+            sage: mat.smith_form(transformation=False)
+            [x + 3     0]
+            [    0     0]
+            [    0     0]
+            sage: rcomp = mat._basis_completion_via_reversed_approx(); rcomp
+            [x + 1   2*x]
+            sage: ccomp = mat.transpose()._basis_completion_via_reversed_approx().transpose(); ccomp
+            [3*x + 1 4*x + 4]
+            [    2*x 5*x + 1]
+            [    6*x       x]
+            sage: rcomp.stack(mat).smith_form(transformation=False)
+            [    1     0]
+            [    0 x + 3]
+            [    0     0]
+            [    0     0]
+            sage: ccomp.augment(mat).smith_form(transformation=False)
+            [    1     0     0     0]
+            [    0     1     0     0]
+            [    0     0 x + 3     0]
+
+        TESTS:
+
+        Corner cases are handled correctly::
+
+            sage: matrix(ring, 0, 0)._basis_completion_via_reversed_approx()
+            []
+            sage: matrix(ring, 0, 2)._basis_completion_via_reversed_approx()
+            [1 0]
+            [0 1]
+            sage: matrix(ring, 2, 0)._basis_completion_via_reversed_approx()
+            []
+        """
+        from sage.matrix.constructor import matrix # for identity
```

Comment:
```suggestion
        from sage.matrix.constructor import matrix  # for identity
```
Two spaces for inline comments ([PEP8](https://peps.python.org/pep-0008/#inline-comments) or `pycodestyle` E225)

Reply:
Oh, I didn't know about this, thanks

Reply:
To be fair, this is not required by the Sage lint CI, which only requires the lint checks specified by `pycodestyle-minimal` [here](https://github.com/sagemath/sage/blob/develop/src/tox.ini#L138). But when possible, new code should follow PEP8 - at least the ones I find nice :)